spinn_machine/virtual_machine.py

```diff
@@ -212,74 +104,56 @@ class _VirtualMachine(object):
-
-        # Version 2/3
-        if version in Machine.BOARD_VERSION_FOR_4_CHIPS:
-            width, height = _verify_4_chip_board(
-                version, width, height, with_wrap_arounds)
-            self._machine = machine_from_size(
-                width, height, origin=self.ORIGIN)
-        # Version 4/5
-        elif version in Machine.BOARD_VERSION_FOR_48_CHIPS:
-            width, height = _verify_48_chip_board(
-                version, width, height, with_wrap_arounds)
-            self._machine = machine_from_size(
-                width, height, origin=self.ORIGIN)
-        # Autodetect
-        elif version is None:
-            _verify_width_height(width, height)
-            if with_wrap_arounds is None:
-                self._machine = machine_from_size(
-                    width, height, origin=self.ORIGIN)
-            elif with_wrap_arounds:
-                self._machine = FullWrapMachine(
-                    width, height, origin=self.ORIGIN)
-            else:
-                self._machine = NoWrapMachine(
-                    width, height, origin=self.ORIGIN)
-        else:
-            raise SpinnMachineInvalidParameterException(
-                "version",
-                version,
-                "The only supported version numbers are 2, 3, 4, 5")
+        _verify_width_height(width, height)
+        self._machine = machine_from_size(width, height, origin=self.ORIGIN)
 
         # Store the details
         self._sdram_per_chip = sdram_per_chip
-        if with_monitors:
-            self._with_monitors = 1
-            self._weird_processor = False
-        else:
-            self._with_monitors = 0
-            self._weird_processor = True
         self._n_cpus_per_chip = n_cpus_per_chip
-        if n_cpus_per_chip != Machine.MAX_CORES_PER_CHIP:
-            self._weird_processor = True
 
         # Store the down items
-        self._down_cores = defaultdict(set)
+        unused_chips = []
+        if down_chips is not None:
+            for down_chip in down_chips:
+                if isinstance(down_chip, IgnoreChip):
+                    if down_chip.ip_address is None:
+                        unused_chips.append((down_chip.x, down_chip.y))
+                else:
+                    unused_chips.append((down_chip[0], down_chip[1]))
+
+        self._unused_cores = defaultdict(set)
         if down_cores is not None:
-            for (x, y, p) in down_cores:
-                self._down_cores[(x, y)].add(p)
-        self._down_links = down_links if down_links is not None else set()
-        if version in Machine.BOARD_VERSION_FOR_4_CHIPS:
-            self._down_links.update(_VirtualMachine._4_chip_down_links)
-        if down_chips is None:
-            down_chips = []
+            for down_core in down_cores:
+                if isinstance(down_core, IgnoreCore):
+                    self._unused_cores[(down_core.x, down_core.y)].add(
+                        down_core.virtual_p)
```

Comment:
This doesn't care about the IP address like the others.  Should it?

Reply:
Is there a use case for supporting ignores in the 127 range?

Reply:
This should at least be consistent with the chips and links cases.

Reply:
This one should then ignore if the IP address is specified

Reply:
agreed fixed